### PR TITLE
VideoPress: add frame selector to Poster panel

### PR DESCRIPTION
**Pull request by retrofox**
This PR adds the video frame selector to the Poster panel.

_**The feature is hidden behind the feature flag.**_

Fixes #

## Proposed changes:

* VideoPress: add frame selector to Poster panel

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

#### Confirm the feature is hidden behind the beta extensions 

Without `beta` extensions | With `beta` extensions
------|--------
Screen Shot 2023-03-24 at 09 35 18|Screen Shot 2023-03-24 at 09 40 17

#### Check the frame selector

* Go to the block editor
* Create/edit a VideoPress video block
* Set a video
* Open the block sidebar 
* Identify the `Poster` panel
* Confirm you see the toggle to switch to pick the poster from video frames
* Confirm the player changes the position while you change the timestamp by using the TimestampControl

https://user-images.githubusercontent.com/77539/227524961-7fe282ed-429d-40f7-abb0-396ef174fa5c.mov

You can take a look a the debug() logs, too

image

### CODE REVIEW

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/style.scss [hunk 1]
```diff
@@ -49,3 +55,49 @@ body.modal-open .poster-panel__dropdown {
...
+.poster-panel__image-wrapper,
+.poster-panel__frame-wrapper {
+	opacity: 0;
+	display: none;
+
+	&.is-selected {
+		opacity: 1;
+		display: block;
+	}
+}
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/style.scss:103]:
```suggestion
}

```

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx [hunk 2]
```diff
@@ -2,23 +2,80 @@
...
+	}
+}
+
+export const isVideoFramePosterEnabled = () =>
+	!! window?.Jetpack_Editor_Initial_State?.available_blocks?.[ 'v6-video-frame-poster' ];
+
+// Global scripts array to be run in the Sandbox context.
+const sandboxScripts = [];
+
+// Populate scripts array with videopresAjaxURLBlob blobal var.
```
Comment (dhasilva) [inline projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx:53]:
```suggestion
// Populate scripts array with videopressAjaxURLBlob blobal var.
```

**Review by dhasilva — APPROVED** [review]
LGTM :rocket:

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx (hunk 2 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx:53]: https://github.com/Automattic/jetpack/pull/29711/commits/6da17723e6d5149e0e83267e31d668dae519f3ed

File: projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/style.scss (hunk 1 above)
Comment (retrofox) [reply projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/style.scss:103]: 4e1affd0478979834b81c32cf98001f9111459fe

**Pull request merged**